extensions/amp-date-display/1.0/date-display.js

```diff
@@ -65,26 +86,52 @@ let EnhancedVariablesV2Def;
  * @param {!DateDisplayDef.Props} props
  * @return {PreactDef.Renderable}
  */
-export function DateDisplay(props) {
-  const {render, children} = props;
-  const data = /** @type {!JsonObject} */ (getDataForTemplate(props));
+export function DateDisplay({
+  datetime,
+  timestampMs,
+  timestampSeconds,
+  displayIn,
+  locale,
+  offsetSeconds,
+  render,
+  children,
+  ...rest
+}) {
+  const epoch = getEpoch(datetime, timestampMs, timestampSeconds);
+  const data = useMemo(
+    () => getDataForTemplate(epoch, displayIn, locale, offsetSeconds),
+    [epoch, displayIn, locale, offsetSeconds]
```

Comment:
```suggestion
  const data = useMemo(
    () =>
      getDataForTemplate(
        getEpoch(datetime, timestampMs, timestampSeconds),
        displayIn,
        locale,
        offsetSeconds
      ),
    [datetime, timestampMs, timestampSeconds, displayIn, locale, offsetSeconds]
```

Reply:
`getEpoch` code will be removed in the https://github.com/ampproject/amphtml/pull/30228. So I'd rather keep this unchanged to simplify the diff.